File: QB_Challenges/PrepInsta/CPP/03_Sum_of_N_Numbers/SumOfN_Recursion.cpp
#include <iostream>
using namespace std;

int SONN(int num){
    if(num == 1){
        return 1;
    }

    return num + SONN(num - 1);
}

int main(){
    int digit;
    cout <<"Enter the number: ";
    cin >> digit;
    cout <<"The sum of 0 to "<<digit<<" numbers is: "<<SONN(digit) <<endl;
    
    return 0;
}

File: QB_Challenges/PrepInsta/01_Getting_Started/03_Sum_of_N_Numbers/Java/SumOfN_Recursion.java
import java.util.Scanner;

public class SumOfN_Recursion {

    public static int SONN(int num) {
        
        if(num == 1){
            return 1;
        }

        return num + SONN(num - 1);

    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number: ");
        int digit = sc.nextInt();
        System.out.println("the sum of first "+digit+" numbers is: "+ SONN(digit));
        sc.close();
    }
    
}
